src/settings/MappingProfiles/MappingProfilesFormContainer/MappingProfilesFormContainer.js

```diff
@@ -53,6 +54,20 @@ export const MappingProfilesFormContainer = props => {
         .find(transformation => selectedTransformation.fieldId === transformation.fieldId),
     ),
   );
+  const [disabledRecordType, setDisabledRecordType] = useState(null);
+  const [initiallyDisabledRecord] = useState(() => {
```

Comment:
Since the field is not going to be updated, use useMemo for that.
```suggestion
  const [initiallyDisabledRecord] = useMemo(() => {
```